### PR TITLE
Fix overlap descriptor set

### PR DESCRIPTION
**Pull request by quehuo01**
Descriptor set update wasn't handling an update larger than the size
of the original binding correctly (Writing multiple bindings with one update).
The specs says "If a binding has a descriptorCount of zero, it is skipped."

Pull request https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2908 has been closed due to my rebase.
Test has been tested with and without the fix. Test is failing without the fix.

### CODE REVIEW

**Comment by quehuo01** [comment]
CC @jeremyg-lunarg

**Review by jeremyg-lunarg — CHANGES_REQUESTED** [review]
Thanks for the fix! The validation change looks good but github CI is failing because:

1.  The test change  commit message needs to start with "tests:"
2.  It doesn't look like this test works with the MockICD. Probably it needs to be guarded with a check against maxInlineUniformBlockSize like in VkLayerTest.InlineUniformBlockEXT

**Comment by quehuo01** [comment]
@jeremyg-lunarg I addressed the requested changes.

**Comment by jeremyg-lunarg** [comment]
It looks like this change is failing in our internal testing, but only on a Galaxy S10:
```
[ RUN      ] VkLayerTest.DescriptorUpdateOfMultipleBindingWithOneUpdateCall
jni/../../tests/vklayertests_descriptor_renderpass_framebuffer.cpp:10263: Failure
Expected equality of these values:
  res
    Which is: -1000069000
  VK_SUCCESS
    Which is: 0
```
Which is a  failing with VK_ERROR_OUT_OF_POOL_MEMORY. I'll try to see if I can find more information about the failure.

**Comment by jeremyg-lunarg** [comment]
I haven't yet been able to figure out why vkAllocateDescriptorSets() is failing for one device.  Here are the full contents of VkPhysicalDeviceInlineUniformBlockPropertiesEXT on 2 of our android CI devices.

Galaxy S10 (fails):
```
	maxInlineUniformBlockSize=65536
	maxPerStageDescriptorInlineUniformBlocks=4
	maxPerStageDescriptorUpdateAfterBindInlineUniformBlocks=4
	maxDescriptorSetInlineUniformBlocks=4
	maxDescriptorSetUpdateAfterBindInlineUniformBlocks=4
```
Pixel 3a (passes):
```
	maxInlineUniformBlockSize=256
	maxPerStageDescriptorInlineUniformBlocks=32
	maxPerStageDescriptorUpdateAfterBindInlineUniformBlocks=32
	maxDescriptorSetInlineUniformBlocks=32
	maxDescriptorSetUpdateAfterBindInlineUniformBlocks=32

```

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp [hunk 1]
```diff
@@ -10155,3 +10157,128 @@ TEST_F(VkLayerTest, ValidateDescriptorBindingUpdateAfterBindWithAccelerationStru
...
+    writeDesc.descriptorType = VK_DESCRIPTOR_TYPE_INLINE_UNIFORM_BLOCK_EXT;
+    writeDesc.dstBinding = 0;
+    writeDesc.dstArrayElement = 0;
+    writeDesc.dstSet = descSet;
+    writeDesc.pNext = &writeInlineUbDesc;
+
+    m_errorMonitor->Reset();
+    m_errorMonitor->ExpectSuccess();
+    vk::UpdateDescriptorSets(m_device->device(), 1, &writeDesc, 0, nullptr);
+    m_errorMonitor->VerifyNotFound();
```
Comment (jeremyg-lunarg) [inline tests/vklayertests_descriptor_renderpass_framebuffer.cpp:10283]: Just noticed that this test leaks handles for descLayout, descPool and descSet. They need to be freed here, or you can use vk_testing::DescriptorSetLayout and the other wrapper objects which will automatically free them.

**Comment by quehuo01** [comment]
The test is trying to allocate 8 inline uniform block in my understanding (2 float) that would explain why it is failing on Samsung S10
Going to change that, and skip the test if "maxPerStageDescriptorInlineUniformBlocks" is less than 2

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp (hunk 1 above)
Comment (quehuo01) [reply tests/vklayertests_descriptor_renderpass_framebuffer.cpp:10284]: I addressed those, and I try to figure out what the issue could be with the S10 but I cannot see why it does fail, as we are within the limit as we've 2 descriptor block each of 4 bytes.

Comment (jeremyg-lunarg) [reply tests/vklayertests_descriptor_renderpass_framebuffer.cpp:10288]: Yep, leak fixes look good.  To get the S10 to pass I just added a test skip if the descriptor set allocation fails.

**Review by jeremyg-lunarg — APPROVED** [review]

**Pull request merged**